jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStore.java

```diff
@@ -122,17 +123,57 @@ public boolean isPassivating()
     @Override
     public Set<String> doGetExpired( Set<String> candidates )
     {
-        if (candidates == null || candidates.isEmpty())
-        {
-            return Collections.emptySet();
-        }
+        Set<String> expired = new HashSet<>();
         
         long now = System.currentTimeMillis();
-        return candidates.stream().filter( candidate -> {
+        expired.addAll( sessionDataMap.values( mapEntry ->  {
```

Comment:
I don't know how hazelcast stores the distributed map, but presumably not every node has an exact copy of the data? This would effectively suck in the full data for every session onto every node, and also inflate that data into a SessionData object that we mostly would throw away immediately - we're only after the ids of the sessions that have timed out at this stage.  So what we would like to do is have hazelcast remotely query, in the most efficient manner possible, for sessions that have long been expired, and then only return us the key of that session. Later on, the scavenger (Housekeeper) will load only the sessions related to those keys.

Reply:
oh noooooo my rebase had an issue I did in a different way. I will do it again.
sorry for that.

Reply:
OK, so should the call be keySet(Predicate) instead of values(Predicate)?

Reply:
@janbartel if I use Predicate on keys I will not have directly the SessionData so I will have to do a lookup to get the object and make some test on getExpiry. Whereas with a Predicate on the values this executed in parallel on the different hazelcast nodes containing the values.

Reply:
Good point. So there's no way to execute the search predicate on the values, but just return the ids then by the sound of it?